helsingborg-stad/app-mitt-helsingborg
# fix(ios): bitwise OR to logical OR (| -> ||) Yoga.cpp patch

## Explain the changes you’ve made

Fixed issue with compilation on newer XCode versions.

## Explain why these changes are made

compability fix for RN < 0.69 and XCode > 14.3. XCode 14.3 + RN 0.68.3 throws compilation error.

## Explain your solution

Added a patch (see [patch-package](https://github.com/ds300/patch-package)) that runs on post-install that changes `|` to `||` for a line in `Yoga.cpp`.

## How to test

Concrete example:

1. Checkout this branch
2. Verify the app still builds (using XCode or CLI)

## Tested environments

- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Notes

* Issue did not affect CI as it is locked to an older/compatible XCode version (for now).
* Solution taken [from here](https://github.com/facebook/react-native/issues/36758#issuecomment-1496210081)
* Upstream fix for newer RN versions [here](https://github.com/facebook/react-native/commit/52d8a797e7a6be3fa472f323ceca4814a28ef596)
